Makefile

```diff
@@ -106,7 +106,10 @@ retool-setup:
 	@which retool >/dev/null 2>&1 || go get github.com/twitchtv/retool
 	@./scripts/retool sync
 
-check: retool-setup check-all check-plugin
+golangci-lint-setup:
+	@which golangci-lint >/dev/null 2>&1 || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.23.7
```

Comment:
I think it can be placed in the `./bin` directory. This directory can also be used as `GOBIN` for tools.go

Reply:
Done. This pr has conflicts with #2196, it would be better to merge #2196 firstly then I will resolve  these conflicts.

Reply:
I think it's better to merge this PR first. If merge #2196 first, it must manage golangci-lint with tools.go, which import several dependencies, and have to remove them after your PR is merged.